src/DotLiquid.Tests/ContextTests.cs

```diff
@@ -785,34 +790,78 @@ public void TestNestedDynamicAsVariable()
             Assert.AreEqual("Hello", _context["dynamic.lambda.name"]);
         }
 
+        private class ExpandoModel
+        {
+            public int IntProp { get; set; }
+            public string StrProp { get; set; }
+            public ExpandoObject Props { get; set; }
+        }
+
+        // Test case for [Issue #350](https://github.com/dotliquid/dotliquid/issues/350)
+        [Test]
+        public void TestNestedExpandoTemplate_Issue350()
+        {
+            var model = new ExpandoModel()
+            {
+                IntProp = 23,
+                StrProp = "from string prop",
+                Props = new ExpandoObject()
+            };
+            var dic = (IDictionary<string, object>)model.Props;
+            dic.Add("Key1", "from expando prop");

```

Comment:
Nit: when two or more arguments have the same type, it is safer to specify the argument names to remove the ambiguity for human reader of the code (compiler won't care, but compiler is not capable to say if the order was invalid).

Reply:
Davi, give it a break, it's clearly a key/value and only one signature exists. Agree with you in general, not in this case.

Reply:
Yes that's fine here.